cmd/helm/package.go

```diff
@@ -121,8 +122,8 @@ func (p *packageCmd) run(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	if filepath.Base(path) != ch.Metadata.Name {
-		return fmt.Errorf("directory name (%s) and Chart.yaml name (%s) must match", filepath.Base(path), ch.Metadata.Name)
+	if !strings.HasSuffix(path, ch.Metadata.Name) {
+		return fmt.Errorf("directory name (%s) and Chart.yaml name (%s) must match", path, ch.Metadata.Name)
```

Comment:
I believe a bandaid for this would be checking "/" + ch.Metadata.Name -- however I'm unsure of any other way to check for correctness in this case than splitting apart the paths in both the chart name and the file name.